prow/tide/README.md

```diff
@@ -0,0 +1,131 @@
+* `target_url`: URL for tide status contexts.
+* `pr_status_base_url`: The base URL for the PR status page. If specified, this URL is used to construct
+   a link that will be used for the tide status context. It is mutually exclusive with the `target_url` field.
+* `max_goroutines`: The maximum number of goroutines spawned inside the component to
+   handle org/repo:branch pools. Defaults to 20. Needs to be a positive number.
+
+### Queries
+
+The `queries` field specifies a list of queries.
+Each query corresponds to a set of **open** PRs as candidates for merging.
+It can consist of the following dictionary of fields:
+
+* `repos`: List of queried repositories.
+* `labels`: List of labels any given PR must posses.
+* `missingLabels`: List of labels any given PR must not posses.
+* `excludedBranches`: List of branches that get excluded when quering the `repos`.
+* `includedBranches`: List of branches that get included when quering the `repos`.
+* `reviewApprovedRequired`: If set, each PR in the query must have review approved.
+
+Under the hood, a query constructed from the fields follows rules described in
+https://help.github.com/articles/searching-issues-and-pull-requests/.
+Therefore every query is just a structured definition of a standard GitHub
+search query which can be used to list mergeable PRs.
+The field to search token correspondence is based on the following mapping:
+
+* `repos` -> `repo:org/repo`
+* `labels` -> `label:lgtm`
+* `missingLabels` -> `-label:do-not-merge`
+* `excludedBranches` -> `-branch:dev`
+* `includedBranches` -> `branch:master`
+* `reviewApprovedRequired` -> `review:approved`
+
+**Important**: Each query must return a different set of PRs. No two queries are allowed to contain the same PR.
+
+Every PR that need to be rebased is filtered from the pool before processing
+
+### Example
+
+```yaml
+tide:
+  merge_method:
+    kubeflow/community: squash
+  target_url: https://prow.k8s.io/tide.html
+  queries:
+  - repos:
+    - kubeflow/community
+    - kubeflow/examples
+    labels:
+    - lgtm
+    - approved
+    missingLabels:
+    - do-not-merge
+    - do-not-merge/hold
+    - do-not-merge/work-in-progress
+    - needs-ok-to-test
+    - needs-rebase
+```
+
+**Explanation**: The component starts periodically querying all PRs in `github.com/kubeflow/community` and
+`github.com/kubeflow/examples` repositories that have `lgtm` and `approved` labels set
+and do not have `do-not-merge`, `do-not-merge/hold`, `do-not-merge/work-in-progress`, `needs-ok-to-test` and `needs-rebase` labels set.
+All PRs that conform to the criteria are processed and merged.
+The processing itself can include running jobs (e.g. tests) to verify the PRs are good to go.
+At the same time all commits in PRs from `github.com/kubeflow/community` repository are squashed before merging.
+
+## Presubmits
+
+Before a PR is merged, Tide ensures that all jobs configured as required in the `presubmits` part of the `config.yaml` file
```

Comment:
Explain how a job is set as required (`skip_report`: `true`)

Reply:
> Explain how a job is set as required (skip_report: true)

Sorry, skip_report=false makes it required.

Reply:
Raither `skip_report: false`?